features/step_definitions/app_steps.rb~

```diff
@@ -0,0 +1,120 @@
+    pending "Unimplemented"
+end
+
+Given /^an arena is set up with posts from (.*) ago$/ do |age_strings|
+    age_strings = age_strings.split(",").map {|c| c.strip}
+    users = [
+        User.create(:email => "alice@example.com", :password => "12345678"),
+        User.create(:email => "bob@example.com", :password => "12345678")
+    ]
+    arena = users[0].arenas.create :user1 => users[0], :user2 => users[1]
+    conversation = arena.conversations.create :title => "From how long ago are you posting?"
+    age_strings.each_with_index do |age, index|
+        value, units = age.split()
+        age = Integer(value).send(units).ago
+        author = users[index % 2]
+        Timecop.freeze age
+        conversation.posts.create :text => "Hello world!", :author => author
+        Timecop.return
+    end
+end
+
+Given /^an arena is set up with posts containing (.*)$/ do |messages|
+    messages = messages.split(",").map do |c|
+        c.strip!
+        c[1...c.length - 1]
+    end
+    users = [
+        User.create(:email => "alice@example.com", :password => "12345678"),
+        User.create(:email => "bob@example.com", :password => "12345678")
+    ]
+    arena = users[0].arenas.create :user1 => users[0], :user2 => users[1]
+    conversation = arena.conversations.create :title => "What messages are you posting?"
+    messages.each_with_index do |message, index|
+        conversation.posts.create :text => message, :author => users[index % 2]
+    end
+end
+
+Given /^an empty arena is set up$/ do
+    first = User.create(:email => "alice@example.com", :password => "password")
+    second = User.create(:email => "bob@example.com", :password => "password")
+    arena = first.arenas.create :user1 => first, :user2 => second
+    conversation = arena.conversations.create :title => "Why do you never post in here?"
+end
+
+# Navigating to page
+
+Given /^I am on (?:the|a|my) (.*?) page$/ do |page_name|
+    case (page_name)
+        when "Login/Register"
+            visit "/login"
+        when "Survey Topic Checkboxes"
+            visit "/login"
+            step 'I fill in "email" with "alice@example.com"'
+            step 'I fill in "password" with "12345678"'
+            step 'I press "Start Registration"'
+            step "And I wait 2 seconds"
+        when "survey"
+            pending "No survey route."
+        when "home"
+            visit "/"
+            step "And I wait 2 seconds"
+        when "profile"
+            visit "/#/profile/1"
+            step "And I wait 2 seconds"
+        when "conversation"
+            pending "No conversation route."
+        else
+            raise "Could not navigate to the #{page_name} page."
+    end
+end
+
+When /^I expand all names in the sidebar$/ do
+    find(".gladiator-heading-container").click
+end
+
+<<<<<<< HEAD
+Given /^I sign in$/ do
+    step 'I sign in as "alice@example.com" with password "12345678"'
+end
+=======
```

Comment:
?? what happened here? If you need reference check out iteration3 and use that.

I'm pretty sure "I sign in" is correct.